src/examples/lock/DeviceController.cpp

```diff
@@ -433,37 +436,56 @@ void DeviceController::FactoryResetButtonHandler()
 }
 
+/**
+ * Sends an IdentifyRequest on a multicast address that reaches Sleepy End Devices (SED).
+ *
+ * Since the multicast must reach Sleepy End Devices, the multicast address is constructed following these guidelines:
+ * - Sleep End Devices subscribe to the mesh local prefix-based multicast addresses (link local and realm local) by default.
+ * - To reach the SED in the realm local, you can use the Realm-Local mesh local prefix-based multicast address which follows these rules according to RFC3306.
+ *     - flags: 3
+ *     - scope: 3
+ *     - plen: mesh local prefix length
+ *     - network prefix: mesh local prefix
+ *     - group id: 1
+ * - For the thread mesh local prefix, it is by default 8 bytes (64bits), and OT provides otThreadGetMeshLocalPrefix()
+ *   to retrieve the mesh local prefix.
+ * - In weave, IPAddress::MakeIPv6PrefixMulticast() method can be used to construct such prefix based multicast addresses.
+ */
 void DeviceController::SendIdentifyRequestButtonHandler()
 {
     WeaveLogProgress(Support, "DeviceController::SendIdentifyRequestButtonHandler()");
-
     DeviceController & _this = GetDeviceController();
-    WEAVE_ERROR err          = WEAVE_NO_ERROR;
-    IdentifyRequestMessage identifyReqMsg;
-    nl::Inet::IPAddress ip_addr;
 
     if (!ConfigurationMgr().IsMemberOfFabric())
     {
         WeaveLogError(Support, "DeviceDiscovery err: Device not fabric provisioned");
         return;
     }
-    uint16_t vendorId;
-    ConfigurationMgr().GetVendorId(vendorId);
-    uint16_t productId;
-    ConfigurationMgr().GetProductId(productId);
-
-    // FIXME: Because these are Sleepy End Devices, mut set the multicast address in a special way.
-    // Waiting for Tread team to provide that information.
-    ip_addr                        = nl::Inet::IPAddress::MakeIPv6WellKnownMulticast(nl::Inet::kIPv6MulticastScope_Realm,
-                                                              nl::Inet::kIPV6MulticastGroup_AllNodes);
+
+    // Get the Thread mesh-local prefix
+    const otMeshLocalPrefix * otMeshPrefix = otThreadGetMeshLocalPrefix(ThreadStackMgrImpl().OTInstance());
+    uint64_t otMeshPrefix64 = nl::Weave::Encoding::BigEndian::Get64(otMeshPrefix->m8);
+
+    // Construct the all-thread-nodes multicast address
+    nl::Inet::IPAddress allThreadNodesAddr = IPAddress::MakeIPv6PrefixMulticast(nl::Inet::kIPv6MulticastScope_Realm, 64, otMeshPrefix64, 1);
+    char addrStr[50];
+    allThreadNodesAddr.ToString(addrStr, sizeof(addrStr));
+    WeaveLogDetail(Support, "All thread nodes multicast address: [%s]", addrStr);
+
+    IdentifyRequestMessage identifyReqMsg;
     identifyReqMsg.TargetFabricId  = ::nl::Weave::DeviceLayer::FabricState.FabricId;
     identifyReqMsg.TargetModes     = kTargetDeviceMode_UserSelectedMode;
-    identifyReqMsg.TargetVendorId  = vendorId;
-    identifyReqMsg.TargetProductId = productId;
+    identifyReqMsg.TargetVendorId  = 0xFFFF; // Any vendor
```

Comment:
Is using 0xFFFF for VID/PID an accepted practice, and documented in an RFC somewhere?  If so, please call it out.

Reply:
It comes from DeviceDescription.cpp. We could file a bug for openweave to define a constant for it.